### ISSUE TITLE
Plain scalars with colons inside flow mappings & sequences are valid YAML

### ISSUE THREAD

**Issue by jstoiko** [go-yaml/yaml#295]
Scalars with colons, such as URLs, when they are inside flow mappings, e.g.
```
a: {b: https://github.com/go-yaml/yaml}
```

and sequences, e.g. 
```
a: [https://github.com/go-yaml/yaml]
```

make `go-yaml` choke with `found unexpected ':'`.

The specs however explicitly allow it:
> Normally, YAML insists the “:” mapping value indicator be separated from the value by white space. A benefit of this restriction is that the “:” character can be used inside plain scalars, as long as it is not followed by white space. This allows for unquoted URLs and timestamps.

On a related note, `libyaml` (the reference parser for YAML in C) had a similar issue but it seems to have been [addressed](https://github.com/yaml/libyaml/pull/28) and about to be released.

**Comment by rogpeppe** [comment]
Fixed in devel branch by PR #305

**Comment by asomov** [comment]
I see no tests added for the failing corner cases:

```
{a: 1}
{a:1}
{a:}
```
